api/src/main/java/com/vinumeris/updatefx/UpdateFX.java

```diff
@@ -85,9 +85,17 @@ public static Path findAppExecutable(Path appInstallDir) throws IOException {
         if (os.contains("mac")) {
             // Returns path to the .app directory, which normally contains only a Contents directory.
             List<Path> exes = Utils.listDir(appInstallDir.resolve("Contents/MacOS"));
-            if (exes.size() != 1)
-                throw new IllegalStateException("Found unknown number of app executables");
-            return exes.get(0);
+            for (Path exe : exes) {
+                /**
+                 * Since 8u40, the packager introduced a "libpackager.dylib"
```

Comment:
Oh sorry, you are right, this is my mistake. I forgot that this code is inside an if (mac) {}

Yes, now this looks good to me. If you fix the typo then I'll merge. Thanks!